aten/src/ATen/native/cpu/UpSampleMoreKernel.cpp

```diff
@@ -14,6 +14,74 @@ namespace {
 
 using scale_t = std::vector<c10::optional<double>>;
 
+template <typename acc_t, typename scalar_t>
+void inline nearest_channels_last_acc(acc_t* gin, scalar_t* gout, int64_t size) {
+  TORCH_CHECK((std::is_same<acc_t, scalar_t>::value),
+              "acc data type of Upsample backward should be same as scalar_t for float or double on CPU.")
+  using Vec = vec::Vectorized<acc_t>;
+  int64_t d = 0;
+  for (; d < size - (size % Vec::size()); d += Vec::size()) {
+    Vec gin_vec = Vec::loadu(gin + d) + Vec::loadu(gout + d);
+    gin_vec.store(gin + d);
+  }
+  for (; d < size; d++) {
+    gin[d] += gout[d];
+  }
+}
+
+template <>
+void inline nearest_channels_last_acc(float* gin, BFloat16* gout, int64_t size) {
+  using bVec = vec::Vectorized<BFloat16>;
+  using fVec = vec::Vectorized<float>;
+  int64_t d = 0;
+  for (; d < size - (size % bVec::size()); d += bVec::size()) {
+    bVec gout_bvec = bVec::loadu(gout + d);
+    fVec gout_fvec0, gout_fvec1;
+    std::tie(gout_fvec0, gout_fvec1) = convert_bfloat16_float(gout_bvec);
```

Comment:
Really minor nit, but you should be able to use structured bindings now that C++17 is fully enabled

Reply:
Use structured bindings.